contracts/ARRRegistry.sol

```diff
@@ -0,0 +1,58 @@
+pragma solidity 0.5.12;
+pragma experimental ABIEncoderV2;
+
+import "@openzeppelin/contracts/ownership/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC721/ERC721Full.sol";
+import "@openzeppelin/contracts/drafts/Counters.sol";
+
+import { IARRRegistry } from "./interfaces/IARRRegistry.sol";
+import { IGovernance } from "./interfaces/IGovernance.sol";
+import { ARRCalculator } from "./ARRCalculator.sol";
+
+/**
+ * @title ARRRegistry
+ * @dev The core registry of any ARR
+ */
+contract ARRRegistry is IARRRegistry, Ownable {
+  using Counters for Counters.Counter;
+
+  event Recorded(uint id, ARR arr);
+
+  IGovernance public _governance;
+
+  Counters.Counter public _id;
+  mapping (uint256 => ARR) public arrs;
+
+  constructor(address owner, IGovernance governance) public {
+    _transferOwnership(owner);
+    _governance = governance;
+  }
+
+  function governance() public view returns (IGovernance) {
+    return _governance;
+  }
+
+  function totalRecords() public view returns (uint) {
+    return _id.current();
```

Comment:
Implies that you are 1 indexing ID? Seems wrong

Reply:
It is, but its the same way the other registry is done

Reply:
Cool, that's fine then I guess